test/tests.js

```diff
@@ -137,26 +121,44 @@ function toggleEnv() {
     process.env.NODE_ENV === 'production' ? 'development' : 'production'
 }
 
-function testEq(exprIn, expect) {
-  const expr = exprIn.replace(/[ \n]+/g, ' ')
-  it(`${expr} => ${show(expect)}`, () => {
-    const actual = run(expr)
+function showFunction(f) {
+  // We print the expression inside a function
+  const functionString = f.toString()
+  // slice `function () {` and the ending `\n}` off the string
+  const bodyString =
+    functionString
+      .slice(functionString.indexOf('{') + 1, functionString.lastIndexOf('}'))
+      .trim()
+  // if the body begins with `return` we removes that as well
+  const withoutReturn =
+    bodyString.startsWith('return') ? bodyString.slice(7) : bodyString
+  return withoutReturn.endsWith(';') ? withoutReturn.slice(0, -1) : withoutReturn
+}
+
+function testEq(expr, expect) {
+  if (typeof expr === "string") {
+    console.log("ERROR: testEq still called with string.")
+    console.log(expr)
+    return
+  }
+  it(`${showFunction(expr)} => ${show(expect)}`, () => {
+    const actual = expr()
     if (!equals(actual, expect))
       throw Error(`Expected: ${show(expect)}, actual: ${show(actual)}`)
 
     toggleEnv()
     try {
-      const actual = run(expr)
+      const actual = expr()
       if (!equals(actual, expect))
         throw Error(`Expected: ${show(expect)}, actual: ${show(actual)}`)
     } finally {
       toggleEnv()
     }
 
-    const exprTy = expr.replace(/\bL\.([a-zA-Z0-9]*)/g, 'T.$1(L.$1)')
-    const typed = run(exprTy)
-    if (!equals(actual, typed))
-      throw Error(`Typed: ${show(typed)}, actual: ${show(actual)}`)
+    // const exprTy = expr.replace(/\bL\.([a-zA-Z0-9]*)/g, 'T.$1(L.$1)')
+    // const typed = run(exprTy)
+    // if (!equals(actual, typed))
+    //   throw Error(`Typed: ${show(typed)}, actual: ${show(actual)}`)
```

Comment:
This will also have to be fixed. What is the purpose of the code?

Reply:
The [types](https://github.com/calmm-js/partial.lenses/blob/master/test/types.js) module defines basic contracts for the library exports.  The above code wraps `L.x` operations in the test code with contract checking code `T.x(L.x)` and then runs the test.  This is somewhat difficult to do with thunks.  One possibility would be to have the types module actually export the library with the contracts in place (e.g. a bit like in [here](https://github.com/polytypic/prettier-printer/blob/master/src/prettier-printer.js) although in PL I don't want to export the contracts as they are for tests), parameterizing the unit test code with the name of the file from which to import the library and then running the tests both with and without the contracts.